app/Decsys/Services/WebhookService.cs

```diff
@@ -21,5 +26,69 @@ IWebhookRepository webhooks
     /// <returns>The created Webhook</returns>
     public int Create(WebhookModel model)
         => _webhooks.Create(model);
+
     
+    /// <summary>
+    /// Triggers webhooks from a given payload.
+    /// </summary>
+    /// <param name="payload">The payload to trigger and Post.</param>
+    public async Task Trigger(PayloadModel payload)
+    {
+        var webhooks = _webhooks.GetWebhooksBySurvey(payload.SurveyId);
+        
+        foreach (var webhook in webhooks)
+        {
+            if (await FilterCriteria(payload, webhook))
+            {
+                await SendWebhook(payload, webhook.CallbackUrl);
+            }
+        }
+    }
+
+    /// <summary>
+    /// Filters the webhooks based on matching criteria.
+    /// Currently focussed on matching page numbers on page navigation.
+    /// </summary>
+    /// <param name="payload">Payload to check</param>
+    /// <param name="webhook">Webhook to match against.</param>
+    /// <returns>True if filter matches</returns>
+    private static Task<bool> FilterCriteria(PayloadModel payload, WebhookModel webhook)
+    {
+        foreach (var eventType in webhook.TriggerCriteria.SelectMany(filter => filter.EventTypes))
+        {
+            switch (payload.EventType)
+            {
+                case PageNavigation pageNavigationEvent:
+                    if (eventType.GetType() == typeof(PageNavigation))
+                    {
+                        var pageEvent = (PageNavigation) eventType;
+                        if (pageNavigationEvent.SourcePage == pageEvent.SourcePage)
+                        {
+                            return Task.FromResult(true);
+                        }
+                    }
+                    break;
+                
+                default:
+                    throw new NotSupportedException($"Event type is not supported.");
+            }
+        }
+
+        return Task.FromResult(false);
+    }
+
+    /// <summary>
+    /// Posts webhook data to a given URL.
+    /// </summary>
+    /// <param name="payload">Payload to Post</param>
+    /// <param name="callbackUrl">Callback url to post to</param>
+    private async Task SendWebhook(PayloadModel payload, string callbackUrl)
+    {
+        var json = JsonConvert.SerializeObject(payload);
+        
+        var content = new StringContent(json, Encoding.UTF8, "application/json");
+        var resp = await _client.PostAsync(callbackUrl, content);
+        resp.EnsureSuccessStatusCode();
```

Comment:
this will crash the calling function and we won't finish writing event logs?

I think a TODO: would be good as we intend to log webhook callback statuses to help people debug (like GitHub does).

Otherwise, for now, silently fail I guess, as we don't wanna crash the calling code.